server/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -760,6 +760,13 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         public String toString() {
             return "store(" + in.toString() + ")";
         }
+
+        @Override
+        public Set<String> getPendingDeletions() throws IOException {
+            // FilterDirectory.getPendingDeletions does not delegate, working around it here.
+            // to be removed once fixed in FilterDirectory.
+            return unwrap(this).getPendingDeletions();
```

Comment:
I think this fix is trappy, but I am ok with it for now. I would like us to delegate all our subclasses of `FilterDirectory` as well. We have a few more. I am torn on the unwrap to be honest.

Reply:
The problem with that route is that most of the subclasses of this are in lucene, including the `MockDirectoryWrapper` created by `LuceneTestCase` and for instance `HardlinkCopyDirectoryWrapper` used by `StoreRecovery`.
I agree that using `unwrap` is questionable, but in the given context this should be safe here until fixed in lucene.

Reply:
I agree, we should still pro-actively implement the right fix in all subclasses and remove the unwrap once lucene is upgraded

Reply:
Thanks @s1monw  for clarifying. Except for two trivial test-only subclasses, I have added the override to all subclasses in 85f2066ec34b1e400d098d3ae0c742ee73288dfe.

Please have another look at your convenience.